compiler/rustc_typeck/src/impl_wf_check.rs

```diff
@@ -207,17 +207,23 @@ fn enforce_impl_params_are_constrained(
 }
 
 fn report_unused_parameter(tcx: TyCtxt<'_>, span: Span, kind: &str, name: &str) {
-    struct_span_err!(
+    let mut err = struct_span_err!(
         tcx.sess,
         span,
         E0207,
         "the {} parameter `{}` is not constrained by the \
         impl trait, self type, or predicates",
         kind,
         name
-    )
-    .span_label(span, format!("unconstrained {} parameter", kind))
-    .emit();
+    );
+    err.span_label(span, format!("unconstrained {} parameter", kind));
+    if kind == "const" {
+        err.note(
+            "proving expressions using a const parameter map each value to a distinct output is not supported",
+        );
```

Comment:
This is worded a little confusingly ... maybe something like this?
```suggestion
        err.note("expressions using a const parameter must map each value to a distinct type");
        err.note("proving this expression has a unique type is not supported");
```

Reply:
Hm I do agree it does need rewording, but the output is not a type, but I agree proving that it's unique is secondary to what `must` be satisfied

Reply:
yeah, I think swapping the error order is better :+1: I do think "map each input to a distinct value" might be slightly clearer :thinking: